src/fatlabel.c

```diff
@@ -116,14 +116,14 @@ static void handle_volid(bool change, bool reset, const char *device, const char
 {
     DOS_FS fs = { 0 };
     char *tmp;
-    unsigned long long conversion;
+    long long conversion;
     uint32_t serial = 0;
 
     if (change) {
 	errno = 0;
-	conversion = strtoull(newserial, &tmp, 16);
+	conversion = strtoll(newserial, &tmp, 16);
 
-	if (!*newserial || *tmp) {
+	if (!*newserial || isspace(*newserial) || *tmp || conversion < 0) {
```

Comment:
why you aren't checking for errno here as well? probably would make sense to factor the checks so they stay consistent across the codebase.

Reply:
Because it is already checked few lined below, see:
https://github.com/dosfstools/dosfstools/pull/92/files#diff-648f9bb939c2d42dafca95e979510a64L134

Reply:
I see, alla in all the patch looks fine even if the checks could be more uniform, but that could be reworked later I guess.

Reply:
ok, thanks for review.